File: server/datasets/controlled_benchmarks/python/multi_student_cohorts/cohort_f2/type2_renamed_34_bst.py
class BinaryElement_34:
    def __init__(self, item_key=0, ptr_l=None, ptr_r=None):
        self.item_key = item_key
        self.ptr_l = ptr_l
        self.ptr_r = ptr_r

class BinarySearchManager_34:
    def __init__(self):
        self.tree_origin = None

    def add_key(self, item_key):
        if not self.tree_origin:
            self.tree_origin = BinaryElement_34(item_key)
            return self.tree_origin
        ptr_walker = self.tree_origin
        while True:
            if item_key < ptr_walker.item_key:
                if not ptr_walker.ptr_l:
                    ptr_walker.ptr_l = BinaryElement_34(item_key)
                    break
                ptr_walker = ptr_walker.ptr_l
            else:
                if not ptr_walker.ptr_r:
                    ptr_walker.ptr_r = BinaryElement_34(item_key)
                    break
                ptr_walker = ptr_walker.ptr_r
        return self.tree_origin

    def contains_key(self, item_key):
        ptr_walker = self.tree_origin
        while ptr_walker:
            if ptr_walker.item_key == item_key:
                return True
            ptr_walker = ptr_walker.ptr_l if item_key < ptr_walker.item_key else ptr_walker.ptr_r
        return False

    def export_inorder(self):
        output_buffer = []
        def depth_scan(element_ptr):
            if element_ptr:
                depth_scan(element_ptr.ptr_l)
                output_buffer.append(element_ptr.item_key)
                depth_scan(element_ptr.ptr_r)
        depth_scan(self.tree_origin)
        return output_buffer

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_f2/type2_renamed_34_bst.java
import java.util.ArrayList;
import java.util.List;

class BinaryElement_34 {
    int itemKey;
    BinaryElement_34 ptrL;
    BinaryElement_34 ptrR;
    BinaryElement_34(int itemKey) { this.itemKey = itemKey; }
}

public class BST {
    BinaryElement_34 treeOrigin;

    public void addKey(int itemKey) {
        if (treeOrigin == null) {
            treeOrigin = new BinaryElement_34(itemKey);
            return;
        }
        BinaryElement_34 ptrWalker = treeOrigin;
        while (true) {
            if (itemKey < ptrWalker.itemKey) {
                if (ptrWalker.ptrL == null) {
                    ptrWalker.ptrL = new BinaryElement_34(itemKey);
                    break;
                }
                ptrWalker = ptrWalker.ptrL;
            } else {
                if (ptrWalker.ptrR == null) {
                    ptrWalker.ptrR = new BinaryElement_34(itemKey);
                    break;
                }
                ptrWalker = ptrWalker.ptrR;
            }
        }
    }

    public boolean containsKey(int itemKey) {
        BinaryElement_34 ptrWalker = treeOrigin;
        while (ptrWalker != null) {
            if (ptrWalker.itemKey == itemKey) return true;
            ptrWalker = (itemKey < ptrWalker.itemKey) ? ptrWalker.ptrL : ptrWalker.ptrR;
        }
        return false;
    }

    public List<Integer> exportInorder() {
        List<Integer> outputBuffer = new ArrayList<>();
        depthScan(treeOrigin, outputBuffer);
        return outputBuffer;
    }

    private void depthScan(BinaryElement_34 elementPtr, List<Integer> outputBuffer) {
        if (elementPtr != null) {
            depthScan(elementPtr.ptrL, outputBuffer);
            outputBuffer.add(elementPtr.itemKey);
            depthScan(elementPtr.ptrR, outputBuffer);
        }
    }
}
